Apply fixture-kit mode to MMMUVLMMixin

majiayu000 opened the pull request:
## Summary
- Split `MMMUVLMTestBase` into separate fixture and mixin classes
- Add `MMMUMixin` for pure MMMU test logic
- Add `MMMUServerBase` in server_fixtures for MMMU-specific server setup
- Keep `MMMUMultiModelTestBase` for backward compatibility with multi-model tests
- Update test files to use the new pattern

## Changes
- `python/sglang/test/kits/mmmu_vlm_kit.py`: Add `MMMUMixin`, rename original to `MMMUMultiModelTestBase`
- `python/sglang/test/server_fixtures/mmmu_fixture.py`: New file with `MMMUServerBase`
- `test/srt/models/test_ministral3_models.py`: Use `MMMUMixin + MMMUServerBase`
- `test/srt/models/test_nvidia_nemotron_nano_v2_vl.py`: Use `MMMUMixin + MMMUServerBase`
- `test/srt/models/test_vlm_models.py`: Use `MMMUMultiModelTestBase`

Fixes #15602

majiayu000 committed:
Apply fixture-kit mode to MMMUVLMMixin

Split MMMUVLMTestBase into:
- MMMUMixin: pure test logic mixin for single-model MMMU tests
- MMMUServerBase: server fixture with MMMU-specific args
- MMMUMultiModelTestBase: for multi-model tests (backward compat)

Updates test files to use the new pattern:
- test_ministral3_models.py: use MMMUMixin + MMMUServerBase
- test_nvidia_nemotron_nano_v2_vl.py: use MMMUMixin + MMMUServerBase
- test_vlm_models.py: use MMMUMultiModelTestBase (multi-model)

Fixes #15602

majiayu000 commented:
Hi maintainers, could you please add the `run-ci` label to trigger CI tests? Thank you!

hnyls2002 committed:
Merge branch 'main' into fix/issue-15602-mmmu-fixture-kit

hnyls2002 commented:
/tag-and-rerun-ci

hnyls2002 commented on python/sglang/test/server_fixtures/mmmu_fixture.py:
```diff
@@ -0,0 +1,76 @@
+        # Set OpenAI API key and base URL environment variables.
+        # Needed for lmms-eval to work.
+        os.environ["OPENAI_API_KEY"] = cls.api_key
+        os.environ["OPENAI_API_BASE"] = f"{cls.base_url}/v1"
```
@majiayu000 Please fix it. Restore it in `tearDownClass`

majiayu000 committed:
chore: retrigger CI

majiayu000 committed:
fix: address review comments

- Use tempfile.TemporaryDirectory() instead of hardcoded ./logs in test_mmmu
- Save and restore OPENAI_API_KEY/OPENAI_API_BASE env vars in tearDownClass
- Add tearDownClass to MMMUMultiModelTestBase for env cleanup

majiayu000 commented:
The CI failures are all from `test_skip_tokenizer_init.py` which is a flaky test unrelated to this PR's changes.

```
FAILED: test_skip_tokenizer_init.py (exit code 1)
requests.exceptions.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
```

All tests related to this PR (lint, stage-a-test-1, accuracy-test-*) have passed. Could you please re-run the failed tests or merge if acceptable?

hnyls2002 committed:
Merge branch 'main' into fix/issue-15602-mmmu-fixture-kit

hnyls2002 committed:
Merge branch 'main' into fix/issue-15602-mmmu-fixture-kit

hnyls2002 committed:
fix

hnyls2002 committed:
Merge branch 'main' into fix/issue-15602-mmmu-fixture-kit

hnyls2002 committed:
fix

hnyls2002 commented:
All related tests have passed (1-gpu) https://github.com/sgl-project/sglang/actions/runs/20550986064/job/59028767281?pr=15615

hnyls2002 commented:
@majiayu000 Kindly remind that the `OPEN_API_KEY` should be set for the client side, not for the server side.

majiayu000 commented:
Thanks for the review and the clarification! 🙏
